tests/trainer/test_trainer.py

```diff
@@ -996,6 +997,49 @@ def test_bnb_compile(self):
             with self.assertRaises(ValueError):
                 _ = Trainer(tiny_model, args, train_dataset=train_dataset)  # noqa
 
+    @require_peft
+    @require_bitsandbytes
+    def test_peft_submodule(self):
+        from peft import LoraConfig, PeftModel, get_peft_model
+
+        # Due to the way the Trainer is implemented we must be able to save the model with 'save_pretrained'
+        # Therefore to use peft submodules you must specify your own 'save_pretrained' method
+        # This example subclass allows for the saving of any and all submodules that are of type PeftModel
```

Comment:
This isn't right as it means we're no longer testing the transformers functionality. Instead, this PR should update `save_pretrained` to enable saving of models with PEFT submodules 

Not sure who's best to tag here @BenjaminBossan @SunMarc

Reply:
I would suggest splitting that issue off into a separate PR. IIUC, this PR is originally about avoiding a false positive `ValueError` that the model is presumably not trainable. I think we should focus on that here.

Now this test touches on another issue, namely that `save_pretrained` of a PEFT model only saves the trainable adapter part, but of the PEFT model itself is just a submodule, `save_pretrained` acts like it would on a normal transformers model and saves the whole checkpoint (unless I misunderstand the intent).

IMO, first of all, this should not be handled here in the test. Instead, I'd just add a comment that says something along the lines of: "Note that save_pretrained saves the whole model here, not just the PEFT adater". The test should still pass, right?

Regarding the question of what the right thing to do is if the model has PEFT submodule(s): IMO this is not quite clear and changing `save_pretrained` for all transformers users to only save the PEFT submodules could be breaking existing code. Probably 90% of users in this situation would only want to save the PEFT adapters, but they would already require special handling to load these submodules correctly (right?) so maybe it's fine that they also need special handling to save them? IMHO, If we want to make it possible to only save the PEFT adapaters of the submodules, it should be configurable, with the default being the status quo.